test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,246 @@
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(AddressZero, accountA.address, depositAmount);
+            });
+
+            it('Should call fallback successfully', async () => {
+                const randomData = ethers.utils.hexlify(
+                    ethers.utils.toUtf8Bytes((Math.random() * 0xfffff).toString(16)),
+                );
+
+                await expect(
+                    accountA.sendTransaction({
+                        to: iexecPoco.address,
+                        value: etherDepositAmount,
+                        data: randomData,
+                    }),
+                )
+                    .to.changeEtherBalances(
+                        [accountA, iexecPoco],
+                        [-etherDepositAmount, etherDepositAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [accountA], [depositAmount])
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(AddressZero, accountA.address, depositAmount);
+            });
+        });
+
+        describe('Deposits', () => {
+            it('Should deposit native tokens', async () => {
+                expect(await iexecPoco.callStatic.deposit(...depositArgs)).to.be.true;
+                await expect(iexecPoco.deposit(...depositArgs))
+                    .to.changeEtherBalances(
+                        [accountA, iexecPoco],
+                        [-etherDepositAmount, etherDepositAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [accountA], [depositAmount])
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(AddressZero, accountA.address, depositAmount);
+            });
+
+            it('Should deposit native tokens for another address', async () => {
+                const depositForArgs = [accountA.address, { value: etherDepositAmount }] as [
+                    string,
+                    { value: BigNumber },
+                ];
+                expect(await iexecPocoAsAdmin.callStatic.depositFor(...depositForArgs)).to.be.true;
+                await expect(iexecPocoAsAdmin.depositFor(...depositForArgs))
+                    .to.changeEtherBalances(
+                        [iexecAdmin, iexecPoco],
+                        [-etherDepositAmount, etherDepositAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [accountA], [depositAmount])
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(AddressZero, accountA.address, depositAmount);
+            });
+
+            it('Should depositForArray with exact value and good array lengths', async () => {
+                const amounts = [depositAmount.mul(2), depositAmount];
+                const totalAmount = getTotalAmount(amounts);
+                const nativeAmountSent = ethers.utils.parseUnits(totalAmount.toString(), 9);
```

Comment:
```suggestion
                const nativeTotalAmount = ethers.utils.parseUnits(totalAmount.toString(), 9);
```
To be iso with `totalAmount`

Reply:
updated